polling_stations/apps/data_finder/helpers/baked_data_helper.py

```diff
@@ -60,12 +90,15 @@ def get_response(self, postcode: Postcode, uprn: Optional[str] = None):
         return result
 
     def get_full_ballots(self, ballot_ids):
-        result = []
-        for ballot_id in ballot_ids:
-            url = ballot_paper_id_to_ee_url(ballot_id)
-            req = session.get(url)
-            result.append(req.json())
-        return result
+        urls = [ballot_paper_id_to_ee_url(ballot_id) for ballot_id in ballot_ids]
+
+        async def fetch_all():
+            async with httpx.AsyncClient() as client:
+                responses = await asyncio.gather(*(fetch(client, url) for url in urls))
+            # filter out any responses that did not return a valid object
+            return [resp for resp in responses if resp is not None]
```

Comment:
This code implements the approach we agreed: If we fail to fetch all the ballots, just roll with the ones that worked. However I do want to make it clear what the consequences are.
Lets say you've got 3 upcoming ballots and 1 request works but the other 2 fail.. if the one that worked was the cancelled ballot, then we're going to plough on as if that was your only ballot. So in that case, `all_ballots_cancelled` would be true and we'll say "your election is cancelled" even though there are actually two you can vote in.
There's probably other variations of that.
I just want to make sure you're aware of/happy with that edge case before we proceed.